integration-tests/src/main/java/com/datastax/oss/quarkus/tests/mapper/InventoryMapper.java

```diff
@@ -22,16 +22,17 @@
 import com.datastax.oss.quarkus.tests.dao.CustomerDao;
 import com.datastax.oss.quarkus.tests.dao.ProductDao;
 import com.datastax.oss.quarkus.tests.dao.ProductReactiveDao;
+import java.util.concurrent.CompletionStage;
 
 @Mapper
 public interface InventoryMapper {
 
   @DaoFactory
-  ProductDao productDao(@DaoKeyspace CqlIdentifier keyspace);
+  CompletionStage<ProductDao> productDao(@DaoKeyspace CqlIdentifier keyspace);
 
   @DaoFactory
-  ProductReactiveDao productReactiveDao(@DaoKeyspace CqlIdentifier keyspace);
+  CompletionStage<ProductReactiveDao> productReactiveDao(@DaoKeyspace CqlIdentifier keyspace);
 
   @DaoFactory
-  CustomerDao customerDao(@DaoKeyspace CqlIdentifier keyspace);
+  CompletionStage<CustomerDao> customerDao(@DaoKeyspace CqlIdentifier keyspace);
```

Comment:
Something about this API has been bothering me since I first saw it but I'm still having trouble articulating exactly what my concern is.  All of the following are mixed together in there:

* Do we really want to expose CompletionStage at this level of the stack?  I completely understand the need to change it at lower levels but it's not immediately clear to me it should surface this high up
* This forces users to interact with the async API for _every_ operation; I'm not sure there's enough benefit there to warrant the intrusion
* After the stage has actually been completed there isn't a huge amount of value in the combination functions offered by CompletionStage.  Since you now have a value (rather than the promise of a future one which may or may not be present) you can manipulate it in any number of ways to achieve similar effects.

So what would the alternatives be?

* You could return the raw value (as was done in the original API) and block if it isn't available yet.  This isn't... great, but will possibly be applicable for a wide number of use cases
** I can easily imagine customers adding blocking methods of these API calls to do just that
* Return a CompletableFuture (rather than a CompletionStage) to offer at least some level of support for both models (blocking vs. async + callback) in a more intuitive way.

I get that this is only a test and that users may implement this differently.  I guess I'm working off of the idea that tests in your repo often serve as a model for how to do things... so I'm wanting to put forward the best example here.

If I'm the only one concerned about this I won't belabor the point, and I certainly don't want to hold up the PR just for this; there's lots of other good stuff in here.  But it did seem worth raising.

Reply:
I'm still skeptical, but at this point we're debating between returning the CompletionStage or the future you get from it... and that's certainly no reason to hold this up.